tests/execute.spec.js

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2016 Resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const m = require('mochainon');
+const _ = require('lodash');
+const os = require('os');
+const childProcess = require('child_process');
+const execute = require('../lib/execute');
+
+describe('Execute', function() {
+
+  describe('.extractAndRun()', function() {
+
+    it('should be able to execute a script', function(done) {
+      const script = _.attempt(() => {
+        if (os.platform() === 'win32') {
+          return {
+            content: '@echo off\necho foo bar baz',
+            originalFilename: 'hello.bat'
+          };
+        }
+
+        return {
+          content: '#!/bin/bash\necho "foo bar baz"',
+          originalFilename: 'hello.sh'
+        };
+      });
+
+      execute.extractAndRun(script, (error, output) => {
+        m.chai.expect(error).to.not.exist;
+
+        if (os.platform() === 'win32') {
```

Comment:
(as long as you add a comment saying that the trim is only to get rid of platform-specific line-endings)